src/dmacro.d

```diff
@@ -251,18 +246,18 @@ public:
                             //printf("\tmacro '%.*s'(%.*s) = '%.*s'\n", m->namelen, m->name, marglen, marg, m->textlen, m->text);
                             marg = memdup(marg, marglen);
                             // Insert replacement text
-                            buf.spread(v + 1, 2 + m.textlen + 2);
+                            buf.spread(v + 1, 2 + m.text.length + 2);
                             buf.data[v + 1] = 0xFF;
                             buf.data[v + 2] = '{';
-                            memcpy(buf.data + v + 3, m.text, m.textlen);
-                            buf.data[v + 3 + m.textlen] = 0xFF;
-                            buf.data[v + 3 + m.textlen + 1] = '}';
-                            end += 2 + m.textlen + 2;
+                            buf.data[v + 3 .. v + 3 + m.text.length] = cast(ubyte[])m.text[];
```

Comment:
Yeah what I meant. Sorry for the wrong suggestion before, I was on the phone which is not the best place write code :D